src/Kiota.Builder/Settings/VsCodeFileManagement.cs

```diff
@@ -0,0 +1,53 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Kiota.Builder.Settings;
+
+public static class VsCodeSettingsManager
+{
+    private static readonly JsonSerializerOptions options = new()
+    {
+        PropertyNamingPolicy = JsonNamingPolicy.CamelCase,
+        WriteIndented = true,
+    };
+    private static readonly SettingsFileGenerationContext context = new(options);
+    public static async Task UpdateFileAsync(string fileUpdate, string fileUpdatePath, string fileUpdateKey, CancellationToken cancellationToken)
+    {
+        ArgumentException.ThrowIfNullOrEmpty(fileUpdate);
+        ArgumentException.ThrowIfNullOrEmpty(fileUpdatePath);
+        ArgumentException.ThrowIfNullOrEmpty(fileUpdateKey);
+        Dictionary<string, object> settings = [];
+
+        // Read existing settings or create new if file doesn't exist
+        if (File.Exists(fileUpdatePath))
+        {
+            using var stream = File.OpenRead(fileUpdatePath);
+            settings = await JsonSerializer.DeserializeAsync(
+                stream,
+                context.DictionaryStringObject,
+                cancellationToken
+            ).ConfigureAwait(false) ?? [];
+        }
+
+        var fileUpdateDictionary = JsonSerializer.Deserialize(fileUpdate, context.DictionaryStringObject);
```

Comment:
Question: is there a specific reason that we need to deserialize the already string after it has just been serialized by the caller into a dictionary again? Would it make sense to simply pass the settings object from the `SettingsFileManagementService.WriteSettingsFileInternalAsync` directly to this function and update the property as needed?

Reply:
The idea was to make the UpdateFileAsync reusable and not just tied to the SettingsFile generation logic